atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/render/TableFactoryRenderer.java

```diff
@@ -51,114 +76,214 @@ public String getClassName() {
-                    line("return new ", TableFactory, "(sharedTriggers, namespace);");
-                } line("}");
-                line();
-                line("public static ", TableFactory, " of(List<Function<? super Transaction, SharedTriggers>> sharedTriggers) {"); {
-                    line("return new ", TableFactory, "(sharedTriggers, defaultNamespace);");
-                } line("}");
-                line();
-                line("private ", TableFactory, "(List<Function<? super Transaction, SharedTriggers>> sharedTriggers, Namespace namespace) {"); {
-                    line("this.sharedTriggers = sharedTriggers;");
-                    line("this.namespace = namespace;");
-                } line("}");
-                line();
-                line("public static ", TableFactory, " of(Namespace namespace) {"); {
-                    line("return of(ImmutableList.<Function<? super Transaction, SharedTriggers>>of(), namespace);");
-                } line("}");
-                line();
-                line("public static ", TableFactory, " of() {"); {
-                    line("return of(ImmutableList.<Function<? super Transaction, SharedTriggers>>of(), defaultNamespace);");
-                } line("}");
-            }
+        JavaFile javaFile = JavaFile.builder(packageName, tableFactory.build())
+                .build();
 
-            private void getTable(String name, TableDefinition table) {
-                String Table = name + "Table";
-                String Trigger = Table + "." + name + "Trigger";
-                if (table.getGenericTableName() != null) {
-                    line("public ", Table, " get", Table, "(Transaction t, String name, ", Trigger, "... triggers) {"); {
-                        line("return ", Table, ".of(t, namespace, name, Triggers.getAllTriggers(t, sharedTriggers, triggers));");
-                    } line("}");
-                } else {
-                    line("public ", Table, " get", Table, "(Transaction t, ", Trigger, "... triggers) {"); {
-                        line("return ", Table, ".of(t, namespace, Triggers.getAllTriggers(t, sharedTriggers, triggers));");
-                    } line("}");
-                }
-            }
+        return javaFile.toString();
+    }
 
-            private void sharedTriggers() {
-                line("public interface SharedTriggers extends");
-                for (String name : definitions.keySet()) {
-                    line("        ", name, "Table.", name, "Trigger,");
-                }
-                replace(",", " {"); {
-                    line("/* empty */");
-                } line("}");
-            }
+    private String getTableName(String name) {
+        return name + "Table";
+    }
+
+    private List<FieldSpec> getFields() {
+        ArrayList<FieldSpec> results = new ArrayList();
+
+        TypeName functionOfTransactionAndTriggersType = ParameterizedTypeName.get(
```

Comment:
Lets maintain the original ordering of fields: defaultNamespace, sharedTriggers, namespace. Same for constructors etc. to keep the diff smaller for generated schemas.

Reply:
Ordering is now maintained in general, except the private constructor which Javapoet always places at the top. Have also changed indentation in the gen code to match the old files in order to reduce diff.